### PR TITLE
Cite-seq datasets

### PR DESCRIPTION
**Pull request by adamgayoso**
Adding some built in data so I can create a tutorial in a cleaner way.

### CODE REVIEW

File: scvi/data/_built_in_data/_cite_seq.py [hunk 1]
```diff
@@ -143,3 +145,80 @@ def _load_spleen_lymph_cite_seq(
...
+    aggregate_proteins: bool = True,
+    mask_protein_batches: int = 0,
+    run_setup_anndata: bool = True,
+):
+    url = "https://ndownloader.figshare.com/files/27458840"
+    save_fn = "pbmc_seurat_v4.h5ad"
+    _download(url, save_path, save_fn)
+    adata = anndata.read_h5ad(os.path.join(save_path, save_fn))
+
+    if aggregate_proteins is True:
```
Comment (watiss) [inline scvi/data/_built_in_data/_cite_seq.py:162]: nit: can remove the `is True`

File: scvi/data/_built_in_data/_cite_seq.py [hunk 2]
```diff
@@ -143,3 +145,80 @@ def _load_spleen_lymph_cite_seq(
...
+        adata.obs["RNA log library size"] = np.log(adata.X.sum(1).A)
+
+        # actually filter
+        adata = adata[adata.obs["Protein log library size"] > 7.6]
+        adata = adata[adata.obs["Protein log library size"] < 10.3]
+        adata = adata[adata.obs["Number proteins detected"] > 150]
+        # filter doublet
+        adata = adata[adata.obs["celltype.l2"] != "Doublet"]
+        # MT
+        adata = adata[adata.obs["pct_counts_mt"] < 12].copy()
```
Comment (watiss) [inline scvi/data/_built_in_data/_cite_seq.py:207]: Why do we make a copy here?

**Review by watiss — APPROVED** [review]

File: scvi/data/_built_in_data/_cite_seq.py (hunk 2 above)
Comment (adamgayoso) [reply scvi/data/_built_in_data/_cite_seq.py:207]: with anndata, without the copy that will be a view of the object in memory, which is not compatible with setup anndata

File: scvi/data/_built_in_data/_cite_seq.py (hunk 1 above)
Comment (adamgayoso) [reply scvi/data/_built_in_data/_cite_seq.py]: fixed

File: scvi/data/_built_in_data/_cite_seq.py (hunk 2 above)
Comment (watiss) [reply scvi/data/_built_in_data/_cite_seq.py:207]: ahhh.
Curiosity question: is this copy an expensive copy the same way as any `adata2 = adata.copy()` is (where adata is _not_ a view)? Or is it a call that doesn't involve any actual byte transfer, but instead just marks the view as a full adata?

Comment (adamgayoso) [reply scvi/data/_built_in_data/_cite_seq.py:207]: this is a good question.... and I don't have the exact answer

**Pull request merged**